bin/aws-kinesis-agent.RedHat

```diff
@@ -142,15 +136,22 @@ do_stop () {
 
     # print status message
     [[ $RETVAL == 0 ]] && success "$DAEMON_NAME shutdown" || failure "$DAEMON_NAME shutdown"
-  ) 9>$MUTEXFILE
+
   RETVAL=$?
-  rm -f $MUTEXFILE
   return $RETVAL
 }
 
+function get_agent_pid() {
+  echo $(ps --ppid $(cat $PIDFILE 2>/dev/null) ho pid 2>/dev/null)
+}
+
 do_status () {
   status -p $PIDFILE $DAEMON_NAME
   RETVAL=$?
+
+  if [[ $RETVAL = 0 && -z $(get_agent_pid) ]]; then
```

Comment:
My bad, it makes sense that we check the process id of start script is live while the process is killed.